pkgs/top-level/all-packages.nix

```diff
@@ -8817,6 +8817,7 @@ in
   protobuf3_0 = lowPrio (callPackage ../development/libraries/protobuf/3.0.nix { });
   # 3.0.0-beta-2 is only introduced for tensorflow. remove this version when tensorflow is moved to 3.0.
   protobuf3_0_0b2 = lowPrio (callPackage ../development/libraries/protobuf/3.0.0-beta-2.nix { });
+  protobuf3_1 = lowPrio (callPackage ../development/libraries/protobuf/3.1.nix { });
```

Comment:
`lowPrio` means that the binaries (in this case protoc) will have a lower prio than those from other store paths if more than one are referenced in the same environment.